lib/bolt/module_installer.rb

```diff
@@ -195,8 +195,10 @@ def install_puppetfile(path, moduledir, config = {})
       @outputter.stop_spin
 
       # Automatically generate types after installing modules
-      @outputter.print_action_step("Generating type references")
-      @pal.generate_types(cache: true)
+      if ok
```

Comment:
When is this false?

Reply:
Anytime installing modules fails. I specifically ran into this when a Puppetfile has duplicate module names -- r10k will error and fail to install modules, but Bolt will continue to try and generate types.

Reply:
It seems like we should raise an error here if `ok` is false in that case.

Reply:
My only hesitation with doing that is we explicitly have a check in the outputter to print a message stating that installing the modules failed https://github.com/puppetlabs/bolt/blob/main/lib/bolt/outputter/human.rb#L428-L434 . I'm fine with making the change, though, if we want to error instead.